vm/src/thread.rs

```diff
@@ -241,6 +241,7 @@ impl RootedThread {
                 stack: Stack::new(),
                 record_map: FieldMap::new(),
                 hook: None,
+                stack_size_limit: VmIndex::max_value(),
```

Comment:
What is the reasoning for having a different field name here? ie. `limit_` vs `max_`?